### DISCUSSION TITLE
PolarFire Soc boot times

### DISCUSSION THREAD

**Discussion by rockoguido** [polarfire-soc/polarfire-soc-documentation#301] [HSS and U-Boot]
Hi.  We have much written about the various boot modes when booting the MSS and we do publish that the boot process of the System Controller and to the point where we release the other cores out of reset and that's about 10ms.    Question is do we publish any other boot times, i.e. booting FreeRTOS from one of the Application Cores or a similar scenario.    
Thx!
Dave

**Comment by vfalanis** [comment]
Hi @rockoguido,

We don't have plans to publish specific boot times as it really depends on various factors such as the specific software release, HSS configuration, operating systems used (Linux, BM, RTOS), boot mechanism (QSPI, SD, mmc), etc. However, there is a very useful feature provided in the HSS called performance counters (CONFIG_DEBUG_PERF_CTRS) that could be used by people interested in measuring boot time.

We did some training in the past on how to use this feature. What would make more sense after looking at this discussion item is to create and publish some base document explaining how to use this feature, so that users can measure their boot times based on their configuration.

**Reply by rockoguido** [reply]
thanks so much and I'll look at the HSS and pursue the use of the performance counters.. !   
RG
